server/paymentProviders/stripe/payment-intent.ts

```diff
@@ -46,11 +55,73 @@ const processOrder = async (order: typeof models.Order): Promise<void> => {
     await order.destroy({ force: true });
     throw new Error(errorMessage);
   }
-};
+}
+
+async function processRecurringOrder(order: typeof models.Order) {
```

Comment:
I feel like there's a huge overlap between the two process functions.
Perhaps we should consolidate both logic in a single `processOrder` function that has conditional calls depending on if it is a subscription or not.

Not something we need to tackle right now, just a tech debt really.

Reply:
Since this is a generic `PaymentIntent` function, perhaps we should also check if the payment method we're using in this order supports or not off-cycle charge.